### PR TITLE
Hydra rebase

### PR DESCRIPTION
**Pull request by maris123**
The project structure is changed and I added Action for enabling/disabling Hydra Settings Panel.

### CODE REVIEW

**Review by dotta — APPROVED** [review]
Reviewed and tested. Looks good!

File: scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      module <- project.scalaModules
+      scalaVersion <- module.sdk.compilerVersion
+    } yield scalaVersion
+
+    if(hydraSettings.isHydraEnabled) {
+      scalaVersions.forall(downloadedScalaVersions.contains(_))
+    } else {
+      true
+    }
+  }
```
Comment (dotta) [inline scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala]: The implementation here got me wonder: what happens if a project has two modules that use two different Scala versions (say 2.12.2 and 2.12.3)?

**Review by dotta — APPROVED** [review]
Code looks good!

File: scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala (hunk 1 above)
Comment (maris123) [reply scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala]: First I get the scala versions from all modules and then I get the already downloaded versions.  Then I look if all the scala versions in the project exist in the downloaded versions.

`scalaVersions.forall(downloadedScalaVersions.contains(_))`

Comment (dotta) [reply scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala]: Let's say that both Hydra for Scala 2.12.2 and 2.12.3 need to be downloaded, by clicking on the "Download" button will the artifacts for both versions be downloaded? If that's the case, it's awesome!

Comment (maris123) [reply scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala]: yes...both versions will be downloaded

Comment (dotta) [reply scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala]: Me happy :)

**Comment by dotta** [comment]
Tested and it works! Happy to merge this if @dragos approves :)

**Comment by dragos** [comment]
Approved! :)

**Pull request merged**